### PR TITLE
Display toast when identity claims reward on behalf of user

### PR DESCRIPTION
**Pull request by sddioulde**
### Description

Display toast and confetti when identity claims reward on behalf of user

- shows see more link in toast if user is not on rewards page
- does not shows see more link in toast if user is on rewards page

### Dragons

Is there anything the reviewer should be on the lookout for? Are there any dangerous changes?

### How Has This Been Tested?

Locally against staging

### How will this change be monitored?

For features that are critical or could fail silently please describe the monitoring/alerting being added.

### CODE REVIEW

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/sd-reward-identity-claim-toast

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/sd-reward-identity-claim-toast

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/sd-reward-identity-claim-toast

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/sd-reward-identity-claim-toast

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/sd-reward-identity-claim-toast

File: src/pages/audio-rewards-page/hooks.ts [hunk 1]
```diff
@@ -19,3 +21,32 @@ export const useOptimisticChallengeCompletionStepCounts = () => {
 
   return completion
 }
+
+// This holds the logic for when a challenge is considered claimable.
+// A challenge is considered claimable when:
+// - it is complete and not disbursed, or
+// - it is undefined (happens during challenge polling) and was previously complete but not disbursed
+export const useCheckClaimable = (
```
Comment (rickyrombo) [inline src/pages/audio-rewards-page/hooks.ts:29]: Why is it undefined during polling?

File: src/pages/audio-rewards-page/components/modals/ChallengeRewards.tsx [hunk 2]
```diff
@@ -253,11 +262,45 @@ const ChallengeRewardsBody = ({ dismissModal }: BodyProps) => {
     }
   }, [dispatch, challenge, specifier])
 
+  const [hasClaimed, setHasClaimed] = useState(false)
+  const previouslyUndisbursedChallenges = useSelector(getUndisbursedChallenges)
+
   useEffect(() => {
```
Comment (piazzatron) [inline src/pages/audio-rewards-page/components/modals/ChallengeRewards.tsx:268]: Could use a comment explaining what this is about, and probably extracting to it's own fn

File: src/pages/audio-rewards-page/store/sagas.ts [hunk 3]
```diff
@@ -239,13 +246,49 @@ export function* watchFetchUserChallenges() {
...
+  const disbursedChallenges = userChallenges
+    .filter(challenge => challenge.is_disbursed)
+    .map(challenge => challenge.challenge_id)
+  const undisbursedChallenges = userChallenges
+    .filter(challenge => !challenge.is_disbursed)
+    .map(challenge => challenge.challenge_id)
+
+  const previouslyUndisbursedChallenges:
+    | ChallengeRewardID[]
+    | null = yield select(getUndisbursedChallenges)
```
Comment (piazzatron) [inline src/pages/audio-rewards-page/store/sagas.ts:267]: `Nullable`

**Review by piazzatron — COMMENTED** [review]
FYI @rickyrombo is going to take over this PR, comments are directed to him

**Pull request closed without merging**